src/decorator/after.decorator.ts

```diff
@@ -0,0 +1,44 @@
+import { AfterMiddleware } from '../middleware';
+import { AfterOption } from '../metadata';
+import { getMetadataArgsStorage } from '../metadata-builder';
+
+/**
+ * Adding ability to declarative override function resolution on schemas with an implementation analog to an express middleware.
+ *
+ * Usage example:
+ * ```
+ * let middleware: AfterMiddleware = (context: any, args: { [key: string]: any }, next: (error?: Error, value?: any) => any): any => {
+ *   if(context.user.role != 'any role') {
+ *     // can use context and resolve/return value as `1000`, for example, regardless of what resolve function actually implements
+ *     next(null, 1000);
+ *   } else {
+ *     // keeps regular resolution flow
+ *     next();
+ *   }
+ * };
+ *
+ * ...
+ *
+ * class ObjType {
+ *   @Field()
+ *   @After(middleware)
+ *   myFunction(input: number): number {
+ *     return input * 2;
+ *   }
+ * }
+ *
+ * ```
+ * @param option Options for an Schema
+ */
+export function After(option: AfterOption | AfterMiddleware) {
```

Comment:
I noticed that this is similar to the `@Before` decorator implementation and I used the `BeforeOption | BeforeMiddleware` there just to keep it compatible with previous implementation but I wouldn't mind changing/breaking it to always be just `BeforeOption`. 

Anyway, I guess we can have just `AfterOption` here to make it similar to other decorator and plan to change the `@Before` to have just `BeforeOption` as argument. What do you think?